changelog_gen/writer.py

```diff
@@ -49,19 +49,19 @@ def consume(self: typing.Self, supported_sections: dict[str, str], sections: Sec
                 continue
 
             header = supported_sections[section]
-            self.add_section(header, {k: v["description"] for k, v in sections[section].items()})
+            self.add_section(header, sections[section])
 
-    def add_section(self: typing.Self, header: str, lines: list[str]) -> None:
+    def add_section(self: typing.Self, header: str, lines: dict[str, dict]) -> None:
```

Comment:
🙃 : maybe `lines` isn't the best name anymore. Should that be `section`? And maybe create a new type for it?
```
# dict [ticket ref -> info]
# info has 2 fields: `description: str` `breaking: bool
SectionData = dict[str, dict[str, str]]
# dict[section key -> section data]
SectionDict = dict[str, SectionData]
```
Maybe even make info a dataclass

Reply:
This becomes `changes: dict[str, list[Change[]` in a later PR, Change being a dataclass. or something similar

Reply:
```    def add_section(self: typing.Self, header: str, changes: dict[str, Change]) -> None:```